pre-contest/c_riddle/sol/sol.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+read L
+read N
+
+declare -A det_cycle
+
+i=0
+while [[ "$i" -lt "$N" ]]
+do
+    ((i+=1))
+    L=$(echo $L | grep -o . | sort | uniq -c | tr -d "[:space:]")
```

Comment:
That line is cool but it **needs to be explained**.
If I understand well, you cut the string in slices of one character, sort the slices, then count adjacent similar lines and remove the spaces.

Reply:
Yes, it's what it does. I commited some comments to the solution.